packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -685,6 +685,13 @@ export class Function extends FunctionBase {
     }
     const architecture = props.architecture ?? (props.architectures && props.architectures[0]);
 
+    if (props.functionName && props.functionName.length>140) {
+      throw new Error('Function name can not be longer than 140 characters.');
+    }
+    if (props.functionName && !this.isValidFunctionName(props.functionName)) {
+      throw new Error('Function name can contain only letters, numbers, hyphens, or underscores with no spaces.');
+    }
```

Comment:
For tokenized values, the validation should not be executed because these characters are allowed in tokens. Condition `!Token.isUnresolved(props.functionName)` should be added.

What do you think of the following suggestion?

```suggestion
    if (props.functionName && !Token.isUnresolved(props.functionName)) {
      if (props.functionName.length>140) {
        throw new Error('Function name can not be longer than 140 characters.');
      }
      if (!this.isValidFunctionName(props.functionName)) {
        throw new Error('Function name can contain only letters, numbers, hyphens, or underscores with no spaces.');
      }
    }
```